platform/core/src/services/MeasurementService/MeasurementService.js

```diff
@@ -184,6 +187,24 @@ class MeasurementService {
     return measurement;
   }
 
+  setMeasurementSelected(measurementUID, selected) {
+    const measurement = this.getMeasurement(measurementUID);
+    if (measurement) {
+      measurement.selected = selected;
+
+      const eventName = this.EVENTS.MEASUREMENT_SELECTED_CHANGED;
+
+      const hasListeners = Object.keys(this.listeners).length > 0;
+      const hasCallbacks = Array.isArray(this.listeners[eventName]);
+
+      if (hasListeners && hasCallbacks) {
+        this.listeners[eventName].forEach(listener => {
+          listener.callback({ measurement });
+        });
+      }
```

Comment:
I see this logic is used in another method, can you refactor it to an internal method? something like `_runCallbacksIfNecessary` or similar